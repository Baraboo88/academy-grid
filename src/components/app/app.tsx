import { ThemeProvider } from 'styled-components';
import {
  Switch,
  Route,
  BrowserRouter as Router,
} from '../common/common';

import DetailedQuest from '../detailed-quest/detailed-quest';
import Contacts from '../contacts/contacts';
import Home from '../home/home';
import { appTheme } from './common';
import * as S from './app.styled';
import React from 'react';
import NotFound from '../not-found/not-found';

const App = () => (
  <ThemeProvider theme={appTheme}>
    <S.GlobalStyle />
    <Router>
      <Switch>
        <Route exact path="/detailed-quest/:id" component={DetailedQuest}/>
        <Route exact path="/contacts">
          <Contacts />
        </Route>
        <Route exact path="/" >
          <Home />
        </Route>
        <Route>
          <NotFound/>
        </Route>
      </Switch>
    </Router>
  </ThemeProvider>
);

export default App;
