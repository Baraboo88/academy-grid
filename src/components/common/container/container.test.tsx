import * as renderer from 'react-test-renderer';
import * as React from 'react';

import { ThemeProvider } from 'styled-components';
import { appTheme } from '../../app/common';

import 'jest-styled-components';

import Container from './container';

it(`Container successfully rendered`, () => {

  const tree = renderer.create(<ThemeProvider
    theme={appTheme}><Container><div/></Container></ThemeProvider>);
  expect(tree).toMatchSnapshot();
});
