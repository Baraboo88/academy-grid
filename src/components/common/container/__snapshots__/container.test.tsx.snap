// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container successfully rendered 1`] = `
.c0 {
  max-width: 1366px;
  width: 100%;
  margin: 0 auto;
}

<div
  className="c0"
>
  <div />
</div>
`;
